src/Durable/OrchestrationInvoker.cs

```diff
@@ -19,6 +20,16 @@ public Hashtable Invoke(OrchestrationBindingInfo orchestrationBindingInfo, IPowe
             try
             {
                 var outputBuffer = new PSDataCollection<object>();
+                var context = orchestrationBindingInfo.Context;
+
+                // context.History should never be null when initializing CurrentUtcDateTime
+                if (context.History != null)
+                {
+                    var orchestrationStart = context.History.FirstOrDefault(
```

Comment:
Similarly, if the history is always supposed to have the OrchestratorStarted event, let's invoke `First` instead of `FirstOrDefault`, so we get an exception right there if this contract is ever broken.